frontend/src/scenes/session-recordings/player/playerUtils.ts

```diff
@@ -176,3 +177,17 @@ export const convertPlayerPositionToX = (
     const playerTime = getPlayerTimeFromPlayerPosition(playerPosition, segments)
     return ((playerTime ?? 0) / durationMs) * containerWidth
 }
+
+export function getRowExpandedState<T extends Record<string, any>>(
+    record: T,
+    recordIndex: number,
+    expandable?: ExpandableConfig<T>,
+    isRowExpandedLocal: boolean = false
+): boolean {
+    return (
+        Number(!!expandable && (!expandable.rowExpandable || expandable.rowExpandable(record, recordIndex))) > 0 &&
+        (!expandable?.isRowExpanded || expandable?.isRowExpanded?.(record, recordIndex) === -1
+            ? isRowExpandedLocal
+            : !!expandable?.isRowExpanded?.(record, recordIndex))
+    )
+}
```

Comment:
Why do we need use this function instead of just checking the local state? Might be worth a comment?

Reply:
This accounts for situations when a user might want to override the local expanded state via the `expandable` prop— this allows this component to be controlled or uncontrolled. I copied this over from LemonTableProps which uses this same logic so I can def drop a comment here explaining where all this comes from.